internal/controller/nonadminbackup_controller.go

```diff
@@ -867,3 +888,33 @@ func updateNonAdminBackupDeleteBackupRequestStatus(status *nacv1alpha1.NonAdminB
 	status.VeleroDeleteBackupRequest.Status = veleroDeleteBackupRequest.Status.DeepCopy()
 	return true
 }
+
+func updateNonAdminBackupPodVolumeBackupStatus(status *nacv1alpha1.NonAdminBackupStatus, podVolumeBackupList *velerov1.PodVolumeBackupList) bool {
+	if status.PodVolumeBackupStatus == nil {
+		status.PodVolumeBackupStatus = []nacv1alpha1.PodVolumeBackupStatus{}
+	}
+
+	updated := false
+	for _, podVolumeBackup := range podVolumeBackupList.Items {
+		alreadyPresent := false
```

Comment:
Lets take a copy and then use it, `podVolumeBackup := podVolumeBackupList.Items[i]`
This will ensure that each backup status pointer correctly references its corresponding instance.

Reply:
actually, I think you were right

Reply:
removed this logic, closing conversation